### PR TITLE
Validate node_labels syntax at role creation time

### PR DESCRIPTION
**Pull request by awly**
Also improve error output and user validation.
Based on missed feedback in https://github.com/gravitational/teleport/pull/4253#pullrequestreview-499061448

### CODE REVIEW

**Review by fspmarshall — APPROVED** [review]

**Review by klizhentas — APPROVED** [review]

**Review by benarent — CHANGES_REQUESTED** [review]
I've copied out these docs changes and will apply them to a new 4.4 docs version, that I'm going to work on today. 

Can you just remove the docs from 4.3 for now?

**Comment by awly** [comment]
@benarent done!

**Review by benarent — APPROVED** [review]

File: lib/utils/parse/parse.go [hunk 1]
```diff
@@ -175,7 +175,12 @@ type Matcher interface {
 //
 // These expressions do not support variable interpolation (e.g.
 // `{{internal.logins}}`), like Expression does.
-func NewMatcher(value string) (Matcher, error) {
+func NewMatcher(value string) (m Matcher, err error) {
+	defer func() {
+		if err != nil {
+			err = trace.WrapWithMessage(err, "see supported syntax at https://gravitational.com/teleport/docs/enterprise/ssh-rbac/#node_labels-syntax")
```
Comment (webvictim) [inline lib/utils/parse/parse.go:181]: This URL doesn't go to a section - presumably this is the docs section you removed that @benarent is going to re-add into the 4.4 docs?

Comment (awly) [reply lib/utils/parse/parse.go]: nice catch, yeah this was the new section.
updated the link to point to the parent section, which is also appropriate (just less targeted).

**Pull request merged**